sdk/attestation/azure-security-attestation/samples/basic-operations/createclient.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief This sample demonstrates creating an attestation client using the Attestation SDK client
+ * for C++.
+ *
+ * @remark The following environment variables must be set before running the sample.
+ * - ATTESTATION_AAD_URL:  Points to an Attestation Service Instance in AAD mode.
+ *
+ * AttestationClient instances are not always authenticated. This sample shows unauthenticated
+ * access to the client.
+ *
+ */
+
+#include <get_env.hpp>
+
+#include <azure/attestation.hpp>
+
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+using namespace Azure::Security::Attestation;
+using namespace Azure::Security::Attestation::Models;
+using namespace std::chrono_literals;
+
+int main()
+{
+  try
+  {
+    AttestationClientOptions clientOptions;
+
+    // Allow up to 10s of time difference between the attestation client and the attestation
+    // service.
+    clientOptions.TokenValidationOptions.ValidationTimeSlack = 10s;
+
+    // create client
+    AttestationClient attestationClient(std::getenv("ATTESTATION_AAD_URL"), clientOptions);
```

Comment:
Customers will get an ugly exception if they forget to set `ATTESTATION_AAD_URL`.  Wondering if you might want to add another `catch (runtime_error_exception e)` to print `you forgot to set an env var for the sample`.  Or if we might want to update this line to first call getenv() and see what it returns like:
auto const url = std::getenv("....");
if(!url) {
cout<< "mising ... env var";
return 1;
}

Reply:
If it matters, this is ugly exception is there for all the samples, not just attestation (just hit this in the vcpkg smoke tests).

Basically the current sample structure is a bit of an anti-pattern. I think the right thing to do is your "we might want to update" suggestion, or to create a method similar to `test_base::GetEnv` for use in samples (putting it in get_env.hpp).

Reply:
Yes, You can update it for Attestation and create an issue for all other sample to follow the same path.
Or you can create an issue to update all samples in the future for this.
Your choise.